code/game/machinery/computer/communications.dm

```diff
@@ -748,6 +749,28 @@
 /obj/machinery/computer/communications/proc/add_message(datum/comm_message/new_message)
 	messages += new_message
 
+/obj/machinery/computer/communications/proc/cross_server(msg)
+	var/comms_key = CONFIG_GET(string/comms_key)
+	if(!comms_key)
```

Comment:
You should check if neither key is set. It seems weird to require the comms key be set even if we only want to talk to insecure comms key servers.